airflow/jobs.py

```diff
@@ -1821,7 +1821,8 @@ def _execute(self):
 
             for ti in run.get_task_instances():
                 # all tasks part of the backfill are scheduled to run
-                ti.set_state(State.SCHEDULED, session=session)
+                if ti.state == State.NONE:
```

Comment:
A very good question. For the use cases I have in mind, anything that is not a success should possibly be rerun. Unfortunately, I could imagine weird cases with BranchOperators and tigger_rules where my proposal would be problematic. We might have to offer in time one of two cases : 1) Rerun all regardless of state. 2) Just fill the holes. But as Max said, 1) can be covered by clear + 2). So LGTM.